cmd/registry/cmd/upload/styleguide.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upload
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"io/ioutil"
+
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+// normalizeArtifactId makes a provided Artifact Id a lower-cased alphanumeric string
+func normalizeArtifactId(resourceName string) string {
```

Comment:
This function is doing more than just doing lower-case conversion. IIUC it is also taking out all the non-alphanumeric chars from the `Name`.  There is already a name validation in place for artifactIDs. 
Example: Trying to upload a manifest with name: `Test-Manifest`
```
$ registry upload manifest cmd/registry/cmd/resolve/testdata/manifest.yaml --project_id=demo
[ 63550255 ] 2021/09/08 13:43:21 uploading projects/demo/locations/global/artifacts/Test-Manifest
[ 63550255 ] 2021/09/08 13:43:21 rpc error: code = InvalidArgument desc = invalid identifier "Test-Manifest": must match "^[a-z0-9-.]+$"
```

Is there a specific reason why we are removing other characters? I think we, currently allow hyphens in the artifact names. A comment explaining why might help.

If we do this, then I suggest we should log this operation happening. The artifact name that we will end up storing in the registry will be different than what the user defined in the configuration, the user should be notified of this change through the logs.

Reply:
Makes sense to me. I was attempting to sanitize a user-provided name (which is possibly invalid) into a valid name. However, I think the method you're describing is more conventional in the registry, which is to simply throw an error if an invalid name is encountered. Fixed in the recent update 👍